From 001ab3069dae5d3f64f57ade6544f9729eb6a382 Mon Sep 17 00:00:00 2001
From: Paolo Valente <paolo.valente@linaro.org>
Date: Tue, 12 Mar 2019 11:29:58 +0200
Subject: [PATCH] lock: bfq: Fix bugs, reduce exec time and boost performance
 [v3]

 Update to [v3]

This is the v3 of the series
https://lkml.org/lkml/2019/3/10/138
that fixes some bug affecting performance, reduces execution time a
little bit, and boosts throughput and responsiveness.

The difference w.r.t. v2 are new fixes to the patch "block, bfq: print
SHARED instead of pid for shared queues in logs", after patient checks
and suggestions by Holger and Oleksandr.

Thanks,
Paolo

Francesco Pollicino (2):
  block, bfq: print SHARED instead of pid for shared queues in logs
  block, bfq: save & resume weight on a queue merge/split

Paolo Valente (7):
  block, bfq: increase idling for weight-raised queues
  block, bfq: do not idle for lowest-weight queues
  block, bfq: tune service injection basing on request service times
  block, bfq: do not merge queues on flash storage with queueing
  block, bfq: do not tag totally seeky queues as soft rt
  block, bfq: always protect newly-created queues from existing active
    queues
  doc, block, bfq: add information on bfq execution time

Signed-off-by: Paolo Valente <paolo.valente@linaro.org>
---
 block/bfq-iosched.h | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/block/bfq-iosched.h b/block/bfq-iosched.h
index f63dc1165b43..972b08758ae4 100644
--- a/block/bfq-iosched.h
+++ b/block/bfq-iosched.h
@@ -1038,13 +1038,12 @@ static inline void bfq_pid_to_str(int pid, char *str, int len)
 #ifdef CONFIG_BFQ_GROUP_IOSCHED
 struct bfq_group *bfqq_group(struct bfq_queue *bfqq);
 
-#define bfq_log_bfqq(bfqd, bfqq, fmt, args...)	do {			\
+#define bfq_log_bfqq(bfqd, bfqq, fmt, args...) do {	\
 	char pid_str[MAX_PID_STR_LENGTH];	\
 	bfq_pid_to_str((bfqq)->pid, pid_str, MAX_PID_STR_LENGTH);	\
-	blk_add_cgroup_trace_msg((bfqd)->queue,				\
-			bfqg_to_blkg(bfqq_group(bfqq))->blkcg,		\
-			"bfq%s%c " fmt, pid_str,			\
-			bfq_bfqq_sync((bfqq)) ? 'S' : 'A', ##args);	\
+	blk_add_trace_msg((bfqd)->queue, "bfq%s%c " fmt, pid_str,	\
+			bfq_bfqq_sync((bfqq)) ? 'S' : 'A',		\
+				##args);	\
 } while (0)
 
 #define bfq_log_bfqg(bfqd, bfqg, fmt, args...)	do {			\
-- 
2.21.0.155.ge902e9bcae

